### PR TITLE
Updated fusionChartDemo to display time taken and num attempts using actual firebase data from fetchDataDemo

### PR DESCRIPTION
**Pull request by bensjx**

### CODE REVIEW

File: src/achievementsApp/reducer.js [hunk 1]
```diff
@@ -25,5 +26,6 @@ export default combineReducers({
   path,
   paths,
   problem,
-  fetchDataDemo
+  fetchDataDemo,
+  fusionChartDemo
```
Comment (FeynmanDNA) [inline src/achievementsApp/reducer.js]: @bensjx the reducers and action creators are pretty the same as the ones in fetchDataDemo? in that case there is no need to duplicate another set of reducer-actions in your Ben folder

File: src/containers/IdeaLab/Ben/fusionChartDemo.js [hunk 2]
```diff
@@ -1,121 +1,502 @@
...
+            </Fragment>
+          )}
+        </Fragment>
       </Fragment>
     );
   }
 }
 
-// Function to seperate the outliers from the list of data
+var timeTakenArray = [];
```
Comment (FeynmanDNA) [inline src/containers/IdeaLab/Ben/fusionChartDemo.js]: put these variables above the react class component

File: src/containers/IdeaLab/Ben/fusionChartDemo.js [hunk 3]
```diff
@@ -1,121 +1,502 @@
...
+var loading = true;
+var numAttemptsArray = [];
+
+var processDataTime = (analyticsData, filteredAnalytics, activitiesData) => {
```
Comment (FeynmanDNA) [inline src/containers/IdeaLab/Ben/fusionChartDemo.js]: use es6 `let const` instead of var

Comment (FeynmanDNA) [reply src/containers/IdeaLab/Ben/fusionChartDemo.js]: and put this function above the react class

File: src/containers/IdeaLab/Ben/fusionChartDemo.js [hunk 4]
```diff
@@ -1,121 +1,502 @@
...
+          )}
+        </Fragment>
       </Fragment>
     );
   }
 }
 
-// Function to seperate the outliers from the list of data
+var timeTakenArray = [];
+var loading = true;
```
Comment (FeynmanDNA) [inline src/containers/IdeaLab/Ben/fusionChartDemo.js]: are you treating `loading` as global variable in this file? can you try incorporate this to the state?

File: src/containers/IdeaLab/Ben/fusionChartDemo.js [hunk 5]
```diff
@@ -1,121 +1,502 @@
...
+
+      let arrayByActivity = Array.from(
+        new Set(arrayCompletedZero.map(item => item.value[2]))
+      ).map(label =>
+        arrayCompletedZero.filter(item => item.value[2] === label)
+      );
+
+      let finalArray = [];
+
+      for (var i = 0; i < arrayByActivity.length; i++) {
```
Comment (FeynmanDNA) [inline src/containers/IdeaLab/Ben/fusionChartDemo.js]: can also try using `map`

**Comment by bensjx** [comment]
Will work on it! Closing this PR for now

**Comment by FeynmanDNA** [comment]
:+1:

**Pull request closed without merging**